modules/core/src/main/java/org/apache/ignite/internal/processors/cache/GridCacheAdapter.java

```diff
@@ -6794,6 +6899,55 @@ public InvokeAllTimeStatClosure(CacheMetricsImpl metrics, final long start) {
         }
     }
 
+    /** */
+    private class ProfileClosure<T> implements CI1<IgniteInternalFuture<T>> {
+        /** */
+        private static final long serialVersionUID = 0L;
+
+        /** Operation type. */
+        private final CacheOperationType op;
+
+        /** Start time in nanoseconds. */
+        private final long start;
+
+        /**
+         * @param op Operation type.
+         * @param start Start time in nanoseconds.
+         */
+        public ProfileClosure(CacheOperationType op, long start) {
+            this.op = op;
+            this.start = start;
+        }
+
+        /** {@inheritDoc} */
+        @Override public void apply(IgniteInternalFuture<T> fut) {
+            try {
+                if (!fut.isCancelled()) {
```

Comment:
Why we ignore cancelled operations?
We should gather statistics about all operations that user initiate no matter if it succeeds or failed.

Reply:
Fixed.